### PR TITLE
Update rsync script and options

### PR DESCRIPTION
**Pull request by santiph**
New `--no-sync=true/false` option added to skip remote synchronisation completely.
Add new error handlers around the configuration file and its cloud options

### CODE REVIEW

File: gulp/rsync.js [hunk 1]
```diff
@@ -3,20 +3,45 @@ var gulp = require('gulp'),
...
-    syncCloudConfig = optional('./cloud.conf.json') || {},
+    syncCloudConfig = optional('./cloud.conf.json'),
     syncCloudID = gutil.env['sync-cloud'] || 'default',
-    syncClouds = syncCloudConfig.clouds || {},
-    syncCloud = syncClouds[syncCloudID];
+    noSync = gutil.env['no-sync'] || false,
+    syncClouds = syncCloudConfig ? syncCloudConfig.clouds : {},
+    syncCloud = syncClouds ? syncClouds[syncCloudID] : undefined,
+
+    skypeRsync = 'Skypping rsync.';
```
Comment (skazi0) [inline gulp/rsync.js]: Should be skip/Skipping.

File: gulp/rsync.js [hunk 2]
```diff
@@ -28,5 +53,6 @@ gulp.task('rsync', function () {
                 recursive: true,
                 silent: !syncCloudConfig.options.verbose
             }));
+
```
Comment (skazi0) [inline gulp/rsync.js]: Why was this blank line added?

**Review by skazi0 — CHANGES_REQUESTED** [review]
I'm not sure you need this much error handling in a script which is only a part of development environment and doesn't go into the product.
IMO, if there is a template for the config file, there's no need to check in the code if the actual config matches the template.

File: gulp/rsync.js (hunk 1 above)
Comment (santiph) [reply gulp/rsync.js]: good catch

File: gulp/rsync.js (hunk 2 above)
Comment (santiph) [reply gulp/rsync.js]: No need. I'll remove it

**Comment by santiph** [comment]
@skazi0 I started adding an option to avoid rsync at all, and then continued handling other scenarios to help to eventually debug, just for fun
btw, I've updated the changes

File: gulp/rsync.js (hunk 1 above)
Comment (skazi0) [reply gulp/rsync.js]: The typo is still there in the variable name.

Comment (santiph) [reply gulp/rsync.js]: Should be updated now

Comment (skazi0) [reply gulp/rsync.js]: I'm not a native english speaker but I believe it should be `skipRsync` not `skipeRsync`.

**Review by skazi0 — APPROVED** [review]

**Pull request merged**